clueless/state.py

```diff
@@ -41,33 +51,63 @@ def sock_addr(self) -> str:
+        
         elif isinstance(message, JoinGame):
-            self.logger.debug(f'Joining game {message.id}')
-            await GAMES[message.id].add_user(self)
+            print("JoinGame instance")
+            if self.game is None:                           # Check whether this player instance is assocaited with a game
+
+                self.logger.debug(f'Player Validated')      
+                self.game = GAMES[message.id]               # Instance of player gets its own game association
+
+                if self.game.id in GAMES:                   # Check whether the game.id exists in dictionary
+                    self.logger.debug(f'Game {message.id} Validated')
+                    await GAMES[message.id].add_user(self)
+                else:
+                    self.logger.debug(f'Game ID {message.id} DOES NOT exist') # RuntimeWarning: coroutine 'Player.send_message' was never awaited
+                    self.send_message(Status("Game ID does NOT exist"))
+            else:
+                self.logger.debug(f'Player is in a game')
+                self.send_message(Status("Player is already in a game"))
+                
         elif isinstance(message, Register):
-            self.logger.debug(f'Registering as {message.display_name} with {message.character}')
-            self.character = message.character
-            self.display_name = message.display_name
-            await self.game.register_user(self, message)
+            print("Register instance")
+            if self.game.id in GAMES and not self.is_registered:    # Not quite sure if this is enough validation to say a player is in a certain game
+                self.logger.debug(f'Registering as {message.display_name} with {message.character}')
+                self.character = message.character                  # Setting the instance character
+                self.display_name = message.display_name            # Setting the instance display_name
+                await self.game.register_user(self, message)        # Communicate with GameState to validate
+            else:
+                self.logger.debug(f'Player is either not in the game or already registered')
+                self.send_message(Status("Player is either not in the game or already registered"))
+
         elif isinstance(message, Complete):
             self.logger.debug(f'Turn completed')
             await self.game.complete_turn(self.character)
         elif isinstance(message, Move):
             self.logger.debug(f'Move to {message.position}')
             await self.game.move_player(self.character, message.position)
+
         elif isinstance(message, Suggest):
             self.logger.debug(f'Suggestion: {message.room} {message.weapon} {message.suspect}')
-            await self.game.suggestion(self.character, message)
+           
+            if self.game.characters[self.character] is self.display_name:    # Validate Player is in this game
```

Comment:
This should just need to make sure that the player is in _a_ game, i.e. that `self.game is not None`. That'll be a common pattern, could refactor actions like this to add a `Player` method along the lines of:
```
async def game_action(self, action, fail):
  if self.game is not None:
    await action(self.game)
  else:
    await self.send_message(Status(fail))
```
so that this entire if/else becomes `await self.game_action(lambda game: game.suggestion(self.character, message), "Cannot make suggestion before joining game")`
Haven't worked with python's async too much, so not sure of the semantics on the lambdas with async, but I'm sure there's a way to do that.